src/Foundatio/Caching/InMemoryCacheClient.cs

```diff
@@ -189,26 +195,33 @@ public Task<int> RemoveByPrefixAsync(string prefix)
 
     internal void RemoveExpiredKey(string key, bool sendNotification = true)
```

Comment:
I tried to only call this if we call a method with a past expiration and we return immediately. Otherwise just let maintenance remove the item to cut down on the number concurrent updates.